modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/v3_0_1/SamlSpIdpConnectionDataUpgradeProcess.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.saml.persistence.internal.upgrade.v3_0_1;
+
+import com.liferay.portal.kernel.io.unsync.UnsyncStringReader;
+import com.liferay.portal.kernel.upgrade.UpgradeException;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.OrderedProperties;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.io.IOException;
+import java.io.StringWriter;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import java.util.Properties;
+
+/**
+ * @author Olivér Kecskeméty
+ */
+public class SamlSpIdpConnectionDataUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		String selectUserAttributeMappingsSQL =
+			"select samlSpIdpConnectionId, userAttributeMappings from " +
+				"SamlSpIdpConnection";
+
+		try (Statement statement = connection.createStatement();
+			ResultSet resultSet = statement.executeQuery(
+				selectUserAttributeMappingsSQL)) {
+
+			while (resultSet.next()) {
+				long samlSpIdpConnectionId = resultSet.getLong(
+					"samlSpIdpConnectionId");
+				String userAttributeMappings = resultSet.getString(
+					"userAttributeMappings");
+
+				Properties userAttributeMappingsProperties =
+					new OrderedProperties();
+
+				if (Validator.isNotNull(userAttributeMappings)) {
+					userAttributeMappingsProperties.load(
+						new UnsyncStringReader(userAttributeMappings));
+				}
+
+				userAttributeMappingsProperties.forEach(
+					(key, value) -> {
+						if (Validator.isNull(value) &&
+							Validator.isNotNull(key)) {
+
+							userAttributeMappingsProperties.replace(key, key);
+						}
+					});
+
+				try (StringWriter stringWriter = new StringWriter()) {
+					userAttributeMappingsProperties.store(stringWriter, null);
+
+					runSQL(
+						StringBundler.concat(
+							"update SamlSpIdpConnection set ",
+							"userAttributeMappings = '",
+							stringWriter.toString(),
+							"' where samlSpIdpConnectionId = ",
+							String.valueOf(samlSpIdpConnectionId)));
+				}
+			}
+		}
+		catch (IOException | SQLException exception) {
+			throw new UpgradeException(exception);
```

Comment:
Hey Oliver. Is it necessary to catch and re-throw as `UpgradeException` here?
It looks like this duplicates logic in the super class: `UpgradeProcess` catches `Throwable`

Reply:
I've included a commit in the follow up PR mentioned below to cover this change. Feel free to drop it 👍